rat/exceptions.py

```diff
@@ -1,20 +1,40 @@
+from .position_and_range import Range
+
+# TODO: These four errors are basically all the same thing. Probably be possible to simplify them somehow
+
+
+class TokenizeError(Exception):
+    def __init__(self, message: str, range: Range):
+        code_string = range.document.split("\n")[range.start.line]
+        pointer_string = " " * range.start.col + "^" + "~" * max(0, range.end.col - range.start.col - 1)
```

Comment:
The range indicator is pretty cool. How would this work on multiline errors?

Reply:
It'll do a bit of awkwardness :D. After looking at how the syntax highlighting works, I'm gonna change how these ranges are built to just be key tokens.

In the PR as is, the range for the `log` expression in `log(x + 1)` would be the full `log(x + 1)` text -- I want it to just be `log` now.

Reply:
(and there's special multiline logic in the language server specification we can use if we ever wanna get there)

Reply:
(and actually I'm not sure how ranges move between expressions -- anyway, this is just putting a bulk of edits here and then I'll change them in the follow up PR with the editor stuff)